frontend/src/stateless/index.ts

```diff
@@ -0,0 +1,471 @@
+ * It is used to create the object store. It is also used to get the result of the
+ * IndexedDB open request. The result is the IndexedDB database. It is used to create
+ * the transaction and the object store. It is also used to get the result of the
+ * IndexedDB add request. The result is the key of the added object.
+ * @see storeStatelessClusterKubeconfig
+ * @see getStatelessClusterKubeConfigs
+ * @see findKubeconfigByClusterName
+ */
+interface DatabaseEvent extends Event {
+  // target is the request that generated the event.
+  target: IDBOpenDBRequest & {
+    // result is the IndexedDB database. It is used to create the transaction and the object store.
+    result: IDBDatabase;
+  };
+}
+
+/**
+ * DatabaseErrorEvent is the event that is fired when the IndexedDB is upgraded.
+ * It is used to get the error of the IndexedDB open request. It is also used to get
+ * the error of the IndexedDB add request.
+ * @see storeStatelessClusterKubeconfig
+ * @see getStatelessClusterKubeConfigs
+ * @see findKubeconfigByClusterName
+ */
+interface DatabaseErrorEvent extends Event {
+  // target is the request that generated the event.
+  target: IDBOpenDBRequest & {
+    // error is the error of the request.
+    error: DOMException | null;
+  };
+}
+
+/**
+ * CursorSuccessEvent is the event that is fired when the IndexedDB is upgraded.
+ * It is used to get the result of the IndexedDB open request. The result is the
+ * cursor. It is used to iterate through the object store. It is also used to get
+ * the result of the IndexedDB add request. The result is the cursor. It is used to
+ * iterate through the object store.
+ * @see getStatelessClusterKubeConfigs
+ * @see findKubeconfigByClusterName
+ * */
+interface CursorSuccessEvent extends Event {
+  // target is the request that generated the event.
+  target: EventTarget & {
+    // result is the cursor. It is used to iterate through the object store.
+    result: IDBCursorWithValue | null;
+  };
+}
+
+/**
+ * KubeconfigObject is the object that is stored in indexDB as string format.
+ * It is a JSON encoded version of the kubeconfig file.
+ * It is used to store the kubeconfig for stateless clusters.
+ * This is basically a k8s client - go Kubeconfig object.
+ * @see storeStatelessClusterKubeconfig
+ * @see getStatelessClusterKubeConfigs
+ * @see findKubeconfigByClusterName
+ */
+interface KubeconfigObject {
+  // apiVersion is the version of the kubeconfig file.
+  apiVersion: string;
+  // kind is the type of the kubeconfig file. It is always 'Config'.
+  kind: string;
+  // clusters is a list of the clusters.
+  clusters: Array<{
+    // name is the name of the cluster.
+    name: string;
+    // cluster is the cluster information.
+    cluster: {
+      server: string;
+    };
+  }>;
+  // users is a list of the users.
+  users: Array<{
+    // name is the name of the user.
+    name: string;
+    // user is the user information.
+    user: {};
```

Comment:
I think user needs a better type.

Reply:
I think, It is not required, since it can be anything. And the use of KubeconfigObject is to get the config loaded in right format. And `jsyaml.load` does the same. It would be unnecessary to add type checks for users here

Reply:
Sure done

Reply:
The user type is documented here: https://kubernetes.io/docs/reference/config-api/kubeconfig.v1/#AuthInfo

Please add that type, or document with a `@todo` comment that it needs to be improved (with the above URL).

Reply:
Done, modified it as per k8s documentation